packages/angular/cli/src/commands/add/cli.ts

```diff
@@ -346,7 +346,17 @@ export default class AddCommandModule
   }
 
   private async getCollectionName(): Promise<string> {
-    const [, collectionName] = this.context.args.positional;
+    let [, collectionName] = this.context.args.positional;
+
+    // The CLI argument may specify also a version, like `ng add @my/lib@13.0.0`,
+    // but here we need only the name of the package, like `@my/lib`
+    try {
+      const packageIdentifier = npa(collectionName);
```

Comment:
We don't need to use NPA for this. This can be done by using `lastIndexOf`

```ts
const versionIdx = collectionName.lastIndexOf('@');
if (versionIdx > 0) {
    collectionName = collectionName.slice(0, versionIdx)
}
```

Reply:
Many thanks for your review @alan-agius4 ! 🙌 

Agreed, we don't need NPM. I've ushed changes now with a code comment on disregarding the edge-case of npm-aliases (that use `@` separator for other purposes).

All that said, IMHO the solution with NPA was cleaner - no custom code to maintain possibly in the future; no need to add code comments to explain assumptions and disregard of edge cases). Also I was inspired by the usage of NPA in the `run()` method. The runtime cost of executing NPA is low, because we call the method `getCollectionName` only once. So I wouldn't mind reverting to the original solution with NPA, if you like it more. Please let me know :) 

Source: https://docs.npmjs.com/cli/v8/commands/npm-install

Reply:
I am okay with using NPA, however it's worth noting that aliases will not work even if we use NPA.

Reply:
Thanks for your reply. I've reverted to NPA solution now.